### PR TITLE
Picker - filter children (for long lists with search)

### LINKED ISSUE #1244: Search in Picker component
### Describe the bug

When we has very much data in Picker component, the Children will be optimized in FlatList which is used in Picker  and searching will not work correctly.  How to resolve this problem?

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Picker - filter children (for long lists with search).
I think this solved the issue, but it has some code duplication with the `Picker.Item`, do you think it can be removed from the item?
Should resolve #1244

## Changelog
Picker - filter children (for long lists with search)

### CODE REVIEW

File: src/components/picker/index.js [hunk 1]
```diff
@@ -271,6 +273,25 @@ class Picker extends Component {
...
+  get children() {
+    const {searchValue} = this.state;
+    const {children, showSearch} = this.props;
+    if (showSearch && !_.isEmpty(searchValue)) {
+      return this.getFilteredChildren(children, searchValue);
+    }
+
+    return children;
+  }
+
```
Comment (ethanshar) [inline src/components/picker/index.js:294]: Looks good! we should remove the search logic we have in PickerItem code since this one covers it.

Comment (M-i-k-e-l) [reply src/components/picker/index.js:299]: Done

File: src/components/picker/PickerItem.js [hunk 2]
```diff
@@ -80,10 +80,6 @@ const PickerItem = props => {
     );
   };
 
-  if (context.showSearch && shouldFilterOut(context.searchValue, itemLabel)) {
```
Comment (ethanshar) [inline src/components/picker/PickerItem.js:83]: You can also stop passing `showSearch` and `searchValue` in the context provided by the Picker

Comment (M-i-k-e-l) [reply src/components/picker/PickerItem.js:83]: Done

**Review by ethanshar — APPROVED** [review]

**Pull request merged**